### DISCUSSION TITLE
We need official python support just like C#

### DISCUSSION THREAD

**Discussion by richard627z** [godotengine/godot-proposals#7511] [Scripting]
It has been discussed a lot about python vs gdscript. We all know that gdscript is inspired by Python. But this does mean that we are locked out of the entire Python ecosystem of packages and libraries. And it will be very cool if python can interact directly with godot engine. On the other hand, people are mor familiar with python, so python support can attract more people. Thanks to [touilleMan's work](https://github.com/touilleMan/godot-python)，we can use python in godot3. So my question is will python become the official support just like C#? And when?

**Comment by Zireael07** [comment]
There are a lot of 3rd party bindings, Python is just one of the options.

AFAIK there are no plans to make any of said 3rd party bindings official ones - the dev team has enough on their plate with C#

**Comment by dsnopek** [comment]
There's a branch of touilleMan's Python bindings for Godot 4, and there's at least one other in-progress Python binding for Godot 4. As the GDExtension eco-system matures, I suspect there will be lots of languages to chose from using community-made extensions :-)

**Comment by richard627z** [comment]
Thanks.

**Comment by fire** [comment]
Here's another godot engine python integration for 4. https://github.com/maiself/godot-python-extension
